apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainProcessorImpl.java

```diff
@@ -1477,8 +1508,14 @@ private void scan(int height, boolean validate, boolean shutdown) {
                 }
                 lastRestoreTime = 0;
             } catch (SQLException e) {
-                dataSource.rollback(false);
-                throw new RuntimeException(e.toString(), e);
+                //if (e.getErrorCode() != 90007) { //The error with code 90007 is thrown when trying to call a JDBC method on an object that has been closed.
+                try {
+                    dataSource.rollback(false);
+                }catch (IllegalStateException ex){
+                    log.error("Error during the Rollback caused by SQL Exception", e);
```

Comment:
Maybe also throw here BlockchainScanException?

Reply:
I guess it's not necessary, the BlockchainScanedException will be thrown from the next statement.